sdk/servicebus/service-bus/src/util/errors.ts

```diff
@@ -73,24 +73,17 @@ export function getSenderClosedErrorMsg(entityPath: string): string {
  */
 export function getReceiverClosedErrorMsg(
   entityPath: string,
-  isClientClosed: boolean,
   sessionId?: string
 ): string {
-  if (isClientClosed) {
-    return (
-      `The client for "${entityPath}" has been closed. The receiver created by it can no longer be used. ` +
-      `Please create a new client using an instance of ServiceBusClient.`
-    );
-  }
```

Comment:
Why are we removing this?

Reply:
The error message `Please create a new client using an instance of ServiceBusClient.` is not applicable since we don't have any QueueClient or SubscriptionClient. 
We can only be throwing connection closed or the receiver closed errors, that makes the `isClientClosed` check irrelevant. 

Even if the client is closed, the message should suggest creating a new receiver using the createReceiver/createSessionReceiver from SBC.

Reply:
Yes, I would expect that we keep the checks and just change the error message.

> Please create a new client using an instance of ServiceBusClient.

would change to

> Please create a new receiver using an instance of ServiceBusClient.

Reply:
But.. users have no idea what a client is. What's the point?

Reply:
Ok, then a few more edits

> The connection for this receiver is closed due to the ServiceBusClient used to create the receiver being closed. Please create a new receiver using a new instance of ServiceBusClient.

Reply:
> The connection for this receiver is closed due to the ServiceBusClient used to create the receiver being closed. Please create a new receiver using a new instance of ServiceBusClient.


No, the connection is not closed here, existing SBClient can be used, no need to create a new SBClient.

From the code that I removed, the client in `isClientClosed` or `create a new client using an instance of ServiceBusClient` is referring to the QueueClient/SubscriptionClient from track 1, not the SBClient.

Reply:
That is because each receiver is getting to have its own client entity context.

We should be passing `this._context.namespace.wasConnectionCloseCalled` instead of  `this._context.isClosed`  to `isClientClosed`. Then we will be tracking the right thing i.e the case when ServiceBusClient is closed but a user tries to use the receiver

Reply:
But... that is already covered under `throwErrorIfConnectionClosed` which is called from `_throwIfReceiverOrConnectionClosed`.
```ts
  private _throwIfReceiverOrConnectionClosed(): void {
    throwErrorIfConnectionClosed(this._context.namespace);
    if (this.isClosed) {
      const errorMessage = getReceiverClosedErrorMsg(this._context.entityPath, this.sessionId!);
      const error = new Error(errorMessage);
      log.error(`[${this._context.namespace.connectionId}] %O`, error);
      throw error;
    }
  }
```

Reply:
Ah! You win.

![](https://media1.giphy.com/media/POH5k3dYNLrqa7SbVa/giphy.gif)